src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -133,7 +133,7 @@ public function supportsClass(string $class)
      *
      * @final
      */
-    public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
+    public function upgradePassword($user, string $newHashedPassword): void
```

Comment:
shouldn't we throw a TypeError if it is not either a UserInterface or a PasswordAuthenticatedUserInterface as we removed the typehint here ?

Reply:
done